RN v12.0.1 - docs preview (#902)
mia-platform/documentation: docs/release-notes/v12.0.1.md

---
id: v12.0.1
title: Version 12.0.1 Release Notes
sidebar_label: v12.0.1
image: "img/release-note-link-preview.png"
---

> _December 1st, 2023_

Author: ```suggestion
_December 12th, 2023_
```

:::info
Mia-Platform Console v12.0.1 is **now in Preview!**

Console SaaS users can try out v12.0.1 latest improvements in Preview! Open a Service Request to ask for the creation of a sandbox Company in case you do not have access to any Company.

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).
:::

## Console

### Improvements

#### MongoDB CRUD section now manages Fast Data `sv-error` collection

Fast Data collection `sv-error` is now visible in the MongoDB CRUD section.

### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

> * a bug has been fixed that causes the user to lose local changes in case of exiting the Design Area after a successful merge process;

Author: ```suggestion
* a bug has been fixed that causes the user to lose local changes in case of exiting the Design Area after a successful merge process
```

> * fixed a bug that, upon failure of namespaces deletion when deleting a project, caused the respective feedback not to be displayed to the user;

Author: ```suggestion
* fixed a bug that, upon failure of namespaces deletion when deleting a project, caused the respective feedback not to be displayed to the user
```

> * fixed a bug that caused a file handling error in console editors.

Author: ```suggestion
* fixed a bug that caused a file handling error in Console editors.
```

* introduce support to version [`v2.0.0`](/fast_data/inputs_and_outputs.md#message-format-v200-1) of Single View Update event.
By default, the service will continue producing events using `v1.0.0`, whereas the new version be opted-in by setting `SV_UPDATE_VERSION` environment variable to `v2.0.0`.  
More details on this feature can be found in the message [documentation paragraph](/fast_data/inputs_and_outputs.md#single-view-update-message)

##### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* Single View `NO_SV_GENERATED` error is now tracked only when generating a Single View record leads to an empty aggregation
results **and** the corresponding record was not previously existing on the database.
This allows to distinguish between _delete_ operations, which are not tracked anymore, from actual aggregation errors.

For more information check out the service [changelog](/runtime_suite/single-view-creator/changelog.md).

## Microfrontend Composer

## Back-Kit Library

The new version `v1.4.9` of the back-kit library is available!

Refer to back-kit [changelog](/microfrontend-composer/back-kit/changelog.md) for finer-grained details of new versions.

> ### Bug Fix

Author: ```suggestion
#### Bug Fix
```

* Writable Views now properly support lookups adopting a _one-to-many_ relationship (multiple values
are returned by the `$lookup` operation in the view aggregation pipeline)

For more information please check out the service [changelog](/runtime_suite/crud-service/changelog.md).

#### Flow Manager Service

The new version `v2.6.3` of the Flow Manager Service is available!

##### New Features

###### Improved schemas of exposed endpoints

All the endpoints exposed by the service now have tag `Flow Manager` in their schema. Moreover, endpoint `/configuration` is now hidden from service Swagger.

###### Redacted Mongo connection

If the service mount a persistency manager of type `mongo`, the property `connectionUri` will now be redacted from logs.

###### Added possibility to deep merge saga metadata upon events reception

Upon events reception, saga metadata can now be deep merged. Moreover, the service can be instructed to perform specific merging behaviors using an optional custom function.

To learn more about this feature, check out the dedicated [documentation page](/runtime_suite/flow-manager-service/30_configuration.md#settings).

## How to update your Console

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `vX.X.X`.

Author: to insert the correct helm chart version

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `vX.X.X`.

Author: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v11.0.4-rc.3`.
```

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v11.0.4-rc.3`.

Author: @fredmaggiowski is this the correct tag for this preview release?
